lib/src/models/parsed_datum.dart

```diff
@@ -14,9 +14,12 @@ sealed class ParsedDatum with _$ParsedDatum {
             throw ValidationException("Datum hash hex must be exactly $datumHashLength characters long.");
           }
         },
-      ParsedDatumInline() => () {},
+      ParsedDatumInline() => () {
+          if (!RegExp(r'^[0-9a-fA-F]+$').hasMatch(thisClass.datumHex)) {
+            throw ValidationException("Datum hex must be a valid hex string.");
+          }
```

Comment:
Hex string validation should come useful in many places. Better to extract this validation as a reusable function in another file.

Reply:
done